src/Adapter/MSTest.TestAdapter/Helpers/FixtureMethodRunner.cs

```diff
@@ -16,6 +16,13 @@ internal static class FixtureMethodRunner
         Action action, CancellationTokenSource cancellationTokenSource, TimeoutInfo? timeoutInfo, MethodInfo methodInfo,
         IExecutionContextScope executionContextScope, string methodCanceledMessageFormat, string methodTimedOutMessageFormat)
     {
+        if (cancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
Don't bother doing any processing if the token is canceled.

Reply:
Do we precisely know exactly when this can happen?

Reply:
With all the refactorings I don't think it's now hit (outside of special race on cancellation).

Reply:
Keeping it for now, I'll try to run more tests to see if we can get rid of it.